test/controllers/tasks/active.officers.enumerations.unit.ts

```diff
@@ -0,0 +1,22 @@
+jest.mock("../../../src/utils/api.enumerations");
```

Comment:
these tests should be in the active.officers.details.controller.unit.ts
If you want to mock a function just for a particular test lifespan you can use 'spy' which gives you teh option to restore the mock to it's original non mocked state

see the task list test as an example - https://github.com/companieshouse/confirmation-statement-web/blob/main/test/controllers/task.list.controller.unit.ts#L137

Reply:
I have tried but it does not seem to work here. I recommend keeping it where it evrything I have tried I get some nasty side effects on the other tests and I really want them all passing.

Reply:
Is this test actually making a call to the 'buildCorporateOfficerList' function in 'active.officers.details.controller.ts'? Because it's the logic in line 56 of that function that we are trying to exercise/test.

Reply:
Yes we should test this more directly, so well need to export this in a certain way so that we can use spy.
However, even with this it'll still have nasty side effects when put into active.officers.details.controller.unit.ts. It seems to be this line:
jest.mock("../../../src/utils/api.enumerations");

it interfers with the other tests this tells me we should not be mixing them up.

Reply:
I'd like to merge this branch in to fix the test - https://github.com/companieshouse/confirmation-statement-web/pull/519
we just have to mock the api enumerations in the controller test and that should fix it.

Reply:
I did think of that but the other tests need to use the actual api enumerations proper; for the corporate officer I am checking that we resolve the chs values to the descriptions. However, I am checing this elsewhere so we could manage without it.

Reply:
the api.enumertions has it's own unit test that can check if descriptions get resolved. As the api.enumerations is a dependency of the controller it's ok for us to mock it out.
I think martin's other comment was that the controller wasn't actually being called from the test.